### PR TITLE
Upgrade aws-sdk-go to v1.18.0

### PR DESCRIPTION
**Pull request by efekarakus**
*Description of changes:* Upgrade and [pin the version](https://golang.github.io/dep/docs/Gopkg.toml.html#version) of aws-sdk-go to 1.18.0

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by efekarakus** [comment]
* Updated with `dep ensure -update github.com/aws/aws-sdk-go`
* Ran `make docker-test` and tests pass 
* Ran `make build` and used the `compose` command to start two containers

File: ecs-cli/Gopkg.toml [hunk 1]
```diff
@@ -25,7 +25,7 @@
 
 [[constraint]]
   name = "github.com/aws/aws-sdk-go"
-  version = "1.15.2"
+  version = "=1.18.0"
```
Comment (PettitWesley) [inline ecs-cli/Gopkg.toml]: Why is there an = sign?

Comment (efekarakus) [reply ecs-cli/Gopkg.toml]: Good question! @SoManyHs and I recently found out that if you don't have an '=' sign,then `dep ensure` will upgrade the version of the dependency up to the next major version. To pin a version number we should use '='.

> When you specify a version without an operator, dep automatically uses the ^ operator by default. 

https://golang.github.io/dep/docs/Gopkg.toml.html#version

Comment (PettitWesley) [reply ecs-cli/Gopkg.toml]: Interesting, good to know 👍

**Pull request merged**